### DISCUSSION TITLE
Is anyone notice sloooooower startup of application after upgrading to NET9?

### DISCUSSION THREAD

**Discussion by dario-l** [dotnet/aspnetcore#58980] [General]
In our solution startup time increased drastically. It takes now longer more than 2 times 😢 

We are just switch `TargetFramework` from `net8.0` to `net9.0`.

**Comment by martincostello** [comment]
Can you provide more specific details? The team won't be able to help with just "it's slower".

**Reply by dario-l** [reply]
In short:
@martincostello for now we are looking for people who also experienced something similar. We do no know is it only some coincidence or something more serious. That's why I'm writing here in Discussions.

More specific:
Under NET8 our application starts in ~4s. After upgrading to NET9 (just change the target framework) startup time increased to more than ~9s. 😞 

For now we do not know where is the problem, is it in our code or inside of any library we use.

When application starting we do pretty standard stuff, eg.: scanning for specific types using Scrutor, etc.

**Reply by davidfowl** [reply]
Maybe it’s scanning more types now.  Hard to know. Do a binary search and comment out pieces of startup until you find the culprit.

**Reply by dario-l** [reply]
@davidfowl Thanks for pointing me to the right direction. 😃 

We do scanning by `Scrutor` and one of ours method use `FromApplicationDependencies`. That method on NET8 took no more than one second but on NET9 almost 5 sec. We are changed that to scanning only for provided assemblies (`FromAssemblies`) and startup time decreased to ~5 sek.

We are just wondering where such a big difference between these two versions comes from? Did the number of dependent assemblies in NET9 increase so much or are dependencies treated differently?

**Reply by davidfowl** [reply]
> We are just wondering where such a big difference between these two versions comes from? Did the number of dependent assemblies in NET9 increase so much or are dependencies treated differently?

that would require somebody doing a detailed analysis. Hard to guess why. Could be more assemblies, could be an issue with the scanning algorithm, could be anything really…

This is why unbounded assembly scanning is bad 😬.

**Reply by dario-l** [reply]
we have just experienced this hard on ourselves 😁
